WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -151,6 +151,52 @@ class PostCoordinator: NSObject {
         AppRatingUtility.shared.incrementSignificantEvent()
     }
 
+    /// Uploads the changes made to the post to the server.
+    ///
+    /// - warning: Work-in-progress (kahu-offline-mode)
+    @discardableResult @MainActor
+    func _update(_ post: AbstractPost, changes: RemotePostUpdateParameters? = nil) async throws -> AbstractPost {
+        let post = post.original ?? post
+        do {
+            let isExistingPost = post.hasRemote()
+            // TODO: Set overwrite to false once conflict resolution support is added
+            try await PostRepository()._save(post, changes: changes, overwrite: true)
+            show(PostCoordinator.makeUploadSuccessNotice(for: post, isExistingPost: isExistingPost))
+            return post
+        } catch {
+            handleError(error, for: post)
+            throw error
+        }
+    }
+
+    private func handleError(_ error: Error, for post: AbstractPost) {
+        guard let topViewController = UIApplication.shared.mainWindow?.topmostPresentedViewController else {
+            return
+        }
+        let alert = UIAlertController(title: Strings.uploadFailed, message: error.localizedDescription, preferredStyle: .alert)
+        if let error = error as? PostRepository.PostSaveError {
+            switch error {
+            case .conflict:
+                // TODO: Show conflict resolution dialog
+                alert.addDefaultActionWithTitle(Strings.buttonOK, handler: nil)
+                break
+            case .deleted:
+                alert.addDefaultActionWithTitle(Strings.buttonOK) { _ in
+                    self.didPermanentlyDelete(post)
+                }
+            }
+        } else {
+            alert.addDefaultActionWithTitle(Strings.buttonOK, handler: nil)
+        }
+        topViewController.present(alert, animated: true)
+    }
+
+    private func didPermanentlyDelete(_ post: AbstractPost) {
```

Comment:
[imo] This might be a bit clearer?
```suggestion
    private func handlePermanentlyDeleted(_ post: AbstractPost) {
```

Reply:
Updated!